src/main/scala/mesosphere/marathon/api/TaskKiller.scala

```diff
@@ -46,8 +45,8 @@ class TaskKiller @Inject() (
           if (wipe) {
             val expunged = await(expunge(foundInstances))
             val killed = await(killService.killInstances(launchedInstances, KillReason.KillingTasksViaApi))
-          } else {
-            if (launchedInstances.nonEmpty) service.killInstances(runSpecId, launchedInstances)
+          } else if (launchedInstances.nonEmpty) {
+            val killed = await(killService.killInstances(launchedInstances, KillReason.KillingTasksViaApi))
```

Comment:
awaiting the future here means we actually wait until the task is reportedly killed. AFAICT, the future will be completed rather immediately if the task in question is unreachable. However, when the task is unreachable but Mesos doesn't report it as such (e.g. due to agent re-registration rate limit), waiting here will result in a request timeout.

It might make more sense to just trigger the kill and not wait.

Reply:
The original code in `MarathonSchedulerActor` also triggered a scale check (`await(schedulerActions.scale(runSpecId))`) resulting in new tasks being started in most cases almost immediately.
Do you want to wait for the periodic scale check here?

Reply:
Also definitely missing explicit logging on what instance Ids are killed here.

Reply:
As mentioned above (on the old code):
> A scale check will be initiated for each task that is reportedly terminal so the schedulerActions.scale(runSpecId) call here was pretty redundant.

To be precide: after the instance has been killed and is expunged, the `ScaleAppUpdateStepImpl` will send a `ScaleRunSpec` to the schedulerActor.